### ISSUE TITLE
How to use setInterval?

### ISSUE THREAD

**Issue by dashed** [David-OConnor/seed#16]
First off, this is a pretty cool Rust web framework! 👍 

------

I was playing with the [quickstart repo](https://github.com/David-OConnor/seed-quickstart), and tried to add a `setInterval` example: 

https://github.com/dashed/seed-quickstart/blob/70b3f2c76e5d205b06adb4734cfcc1a08f412c5c/src/lib.rs#L42-L57

```rust
fn view(state: seed::App<Msg, Model>, model: Model) -> El<Msg> {

    let callback = Closure::wrap(Box::new(move || {
        state.update(Msg::Increment);
    }) as Box<dyn Fn()>);

    div![
        did_mount(move |_| {
            let window = web_sys::window().unwrap();
            window
                .set_interval_with_callback_and_timeout_and_arguments_0(
                    // Note this method call, which uses `as_ref()` to get a `JsValue`
                    // from our `Closure` which is then converted to a `&Function`
                    // using the `JsCast::unchecked_ref` function.
                    callback.as_ref().unchecked_ref(),
                    1_000,
                )
                .unwrap();
        }),
        button![
            simple_ev("click", Msg::Increment),
            format!("Hello, World × {}", model.val)
        ]
    ]
}
```

But I'm getting this error:

<img width="1680" alt="screen shot 2018-12-31 at 12 39 55 pm" src="https://user-images.githubusercontent.com/139499/50565455-f5677200-0cfc-11e9-9458-1fa544c128d2.png">

I'm pretty much new to WASM, and I'm unsure how to debug this 😞; I'm hoping if there's some guidance to this.

**Comment by David-OConnor** [comment]
This is a bug in the framework - I'll update once fixed.

**Comment by David-OConnor** [comment]
I got this working with a refactoring of your view func:
```rust
fn view(state: seed::App<Msg, Model>, model: Model) -> El<Msg> {  
    div![
        did_mount(move |_| {
            let state2 = state.clone();

            let callback = Closure::wrap(Box::new(move || {
                state2.update(Msg::Increment);
            }) as Box<dyn Fn()>);
            
            let window = web_sys::window().unwrap();
            window
                .set_interval_with_callback_and_timeout_and_arguments_0(
                    // Note this method call, which uses `as_ref()` to get a `JsValue`
                    // from our `Closure` which is then converted to a `&Function`
                    // using the `JsCast::unchecked_ref` function.
                    callback.as_ref().unchecked_ref(),
                    1_000,
                )
                .unwrap();
                callback.forget();
        }),
        button![
            simple_ev("click", Msg::Increment),
            format!("Hello, World × {}", model.val)
        ]
    ]
}
```
The key differences are moving the callback into the did_mount closure, using a clone of state to avoid closure/moving problems, and forgetting the callback. I'm not sure if the best approach here is to make a change in the library to facilitate doing things like this, or to update the docs.

**Comment by David-OConnor** [comment]
This will work in the next version:
```rust
fn view(state: seed::App<Msg, Model>, model: Model) -> El<Msg> {  
    div![
        did_mount(move |_| {
            let state2 = state.clone();
            let callback = move || {
                state2.update(Msg::Increment);
            };
            seed::set_interval(Box::new(callback), 1000);
        }),

        button![
            simple_ev("click", Msg::Increment),
            format!("Hello, World × {}", model.val)
        ]
    ]
}
```

**Comment by David-OConnor** [comment]
Published in v0.2.2
